src/jltypes.c

```diff
@@ -2807,10 +2807,6 @@ void jl_init_types(void) JL_GC_DISABLED
     jl_datatype_type->ismutationfree = 1;
     ((jl_datatype_t*)jl_type_type->body)->ismutationfree = 1;
 
-    // Technically not ismutationfree, but there's a separate system to deal
-    // with mutations for global state.
-    jl_module_type->ismutationfree = 1;
```

Comment:
What about `Some{Module}`?

Reply:
Ah, right.. I found it a bit weird to mark `Module` as `ismutationfree`, but I think we have to do so.